### DISCUSSION TITLE
ResponseModel does not force model

### DISCUSSION THREAD

**Discussion by datenwort** [fastapi/fastapi#7345] [Questions] [answered]
Is there a way to force the the response to be equal the response_model?

```
class Item(BaseModel):
    id: int
    shortcut: str
    name: str

@app.get("/items/{item_id}", response_model=Item)
async def get_all_items(item_id: int):
    """Return all assocations."""
    item = {"name": "name", "size": "size"}
    return JSONResponse(status_code=status.HTTP_200_OK, content=item)
```

I would expect, that the response failed cause it is not aligned with the model. Any advice?

**Comment by Kludex** [comment, marked as answer]
You should not return a `JSONResponse` instance if you want FastAPI to handle your models. 
Remember that `get` is a decorator which already uses `JSONResponse` (to be honest is not a `JSONResponse` but is a parent or child class if I remember correctly). How we handle it in FastAPI:
```
class Item(BaseModel):
    id: int
    shortcut: str
    name: str

@app.get("/items/{item_id}", response_model=Item)
async def get_all_items(item_id: int):
    """Return all assocations."""
    item = {"name": "name", "size": "size"}
    return item
```
But this will raise an error. Can you see why? 

**SPOILER ALERT!!!**
Because you have the field `shortcut` and `id` are required ones!!! How do we solve it? There are several ways, I'll show you the simplest for me:
```
class Item(BaseModel):
    id: Optional[int]
    shortcut: Optional[str]
    name: str

@app.get("/items/{item_id}", response_model=Item)
async def get_all_items(item_id: int):
    """Return all assocations."""
    item = {"name": "name", "size": "size"}
    return item
```

**Comment by datenwort** [comment]
As I mentioned, I was expecting the error. Thanks for clarify how to return it.
Will add the status code with the response object instead.

**Comment by tiangolo** [comment]
Thanks for the help here @Kludex ! :clap: :bow: 

Thanks for reporting back and closing the issue @datenwort  :+1:
